aguaclara/design/floc.py

```diff
@@ -1,307 +1,272 @@
+       :param  downstream_H: Desired depth at the downstream end.
+       :type  downstream_H: float * u.m
+       :param  entrancetank_L: Estimated length of entrance tank.
+       :type  entrancetank_L: float * u.m
+       :param  max_W: Maximum width of flocculator based on baffle sheet width.
+       :type  max_W: float * u.m
+       :returns: object
+       :rtype: Flocculator
+       """
+       self.Q = Q
+       self.temp = temp
+       self.max_L = max_L
+       self.Gt = Gt
+       self.HL = HL
+       self.downstream_H = downstream_H
+       self.entrancetank_L = entrancetank_L
+       self.max_W = max_W
+       self.drain_t = drain_t
+
+
+   @property
+   def vel_grad_avg(self):
+       """Calculate the average velocity gradient (G-bar) of water flowing
+       through the flocculator.
+       :returns: Average velocity gradient (G-bar)
+       :rtype: float * 1 / second
+       """
+       return ((u.standard_gravity * self.HL) /
+               (pc.viscosity_kinematic(self.temp) * self.Gt)).to(u.s ** -1)
+
+   @property
+   def retention_time(self):
+       """Calculates the hydraulic retention time neglecting the volume created by head loss in the flocculator.
+       :returns: Hydraulic retention time (:math:`\theta`)
+       :rtype: float * second
+       """
+       return (self.Gt / self.vel_grad_avg).to(u.s)
+
+   @property
+   def vol(self):
+       """Calculate the target volume (not counting the volume added by head loss) of the flocculator.
+       :returns: Target volume
+       :rtype: float * meter ** 3
+       """
+       return (self.Q * self.retention_time).to(u.m ** 3)
+
+
+   @property
+   def W_min_HS_ratio(self):
+       """Calculate the minimum flocculator channel width, given the minimum
+       ratio between expansion height (H) and baffle spacing (S).
+       :returns: Minimum channel width given H_e/S
+       :rtype: float * centimeter
+       """
+       return ((self.HS_RATIO_MIN * self.Q / self.downstream_H) *
+               (self.BAFFLE_K /
+                (2 * self.downstream_H * pc.viscosity_kinematic(self.temp) * self.vel_grad_avg ** 2)) ** (1/3)
+               ).to(u.cm)
+
+
+   @property
+   def channel_n(self):
+       """Calculate the minimum number of channels based on the maximum
+       possible channel width and the maximum length of the channels.
+       Round up to the next even number (factor of 2 shows up twice in equation)
+       The channel width must be greater than the hydraulic width that ensure baffle overlap.
+       Based on the equation for the flocculator volume
+       volume = ([max_L*channel_n] - entrancetank_L)*max_W * downstream_H
+       :returns: number of channels
+       :rtype: float * dimensionless
+       """
+       min_hydraulic_W = np.amax(np.array([1,(self.max_W/self.W_min_HS_ratio).to(u.dimensionless)])) * self.W_min_HS_ratio
+       return 2*np.ceil(((self.vol/(min_hydraulic_W * self.downstream_H) + self.entrancetank_L)/(2*self.max_L)).to(u.dimensionless))
```

Comment:
I wonder if separating this equation into smaller equations with vars that have descriptive names could make this more reasonable. As it stands, this is hard to parse and read through. For instance: 

```
ratio_of_widths = (self.max_W/self.W_min_HS_ratio).to(u.dimensionless)
# 1 is the lower bound because the width can't get smaller than the min
optimal_width_ratio = np.amax(np.array([1,ratio_of_widths]))
min_hydraulic_W = optimal_width_ratio * self.W_min_HS_ratio
length_without_headloss_volume = self.vol/(min_hydraulic_W * self.downstream_H) + self.entrancetank_L
ideal_number_of_channels = (length_without_headloss_volume)/(self.max_L)
actual_even_number_of_channels = 2*np.ceil((ideal_number_of_channels/2).to(u.dimensionless))
return actual_even_number_of_channels
```

Reply:
The idea is that the variable names provide the documentation. This eliminates the need for comments that often get way out of date. And since we're within the scope of the function, you can use nice long and descriptive var names, until it almost reads like english. That is the goal of all code: to be as readable as possible